File: programmers/짝지어 제거하기.py
def solution(s):
    stack = []
    for i in range(len(s)):
        if not stack: # stack이 비어있을 경우
            stack.append(s[i])
        else:
            if s[i] == stack[-1]: # 새로운 s 문자열이 이전의 문자열과 같을 경우
                stack.pop()
            else:
                stack.append(s[i])
    if stack:
        return 0
    else:
        return 1

File: programmers/level2/짝지어_제거하기.cpp
#include <string>
#include <stack>
using namespace std;

int solution(string s)
{
    stack<char> same;
    for (char c : s) {
        if (!same.empty() && same.top() == c)
            same.pop(); // 같은 문자 제거
        else
            same.push(c);
    }

    return same.empty() ? 1 : 0;
}